File: src/com/lordsutch/android/signaldetector/SettingsFragment.java
package com.lordsutch.android.signaldetector;

import android.content.Context;
import android.os.Bundle;
import android.preference.Preference;
import android.preference.Preference.OnPreferenceClickListener;
import android.preference.PreferenceFragment;
import android.webkit.WebView;
import android.widget.Toast;

import java.io.File;
import java.util.Arrays;
import java.util.List;

public class SettingsFragment extends PreferenceFragment {
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Load the preferences from an XML resource
        addPreferencesFromResource(R.xml.preferences);

        Preference myPref = findPreference("clear_cache");
        myPref.setOnPreferenceClickListener(preference -> {
            Context context = getActivity();
            CharSequence text = context.getString(R.string.map_tile_cache_cleared);
            int duration = Toast.LENGTH_SHORT;

            WebView webView = new WebView(context);
            webView.clearCache(true);

            Toast.makeText(context, text, duration).show();
            return false;
        });

        myPref = findPreference("clear_log");
        myPref.setOnPreferenceClickListener(preference -> {
            Context context = getActivity();
            CharSequence text = context.getString(R.string.logFilesToast);
            int duration = Toast.LENGTH_SHORT;

            File logPath = context.getExternalFilesDir("");
            List<String> logNames = Arrays.asList("cellinfolte.csv", "ltecells.csv",
                    "esmrcells.csv", "cdmacells.csv", "gsmcells.csv");

            for (String fileName : logNames) {
                File logFile = new File(logPath, fileName);

                if (logFile.exists()) {
                    boolean deleted = logFile.delete();
                }
            }

            Toast.makeText(context, text, duration).show();
            return false;
        });

    }
}

File: src/com/lordsutch/android/signaldetector/SettingsFragment.kt
package com.lordsutch.android.signaldetector

import android.os.Bundle
import android.webkit.WebView
import android.widget.Toast
import androidx.preference.PreferenceFragmentCompat
import java.io.File

class SettingsFragment : PreferenceFragmentCompat() {
    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        // Load the preferences from an XML resource
        addPreferencesFromResource(R.xml.preferences)

        var myPref = findPreference("clear_cache")
        myPref.setOnPreferenceClickListener {
            val context = activity
            val text = context?.getString(R.string.map_tile_cache_cleared)
            val duration = Toast.LENGTH_SHORT

            val webView = WebView(context)
            webView.clearCache(true)

            Toast.makeText(context, text, duration).show()
            false
        }

        myPref = findPreference("clear_log")
        myPref.setOnPreferenceClickListener {
            val context = activity
            val text = context?.getString(R.string.logFilesToast)
            val duration = Toast.LENGTH_SHORT

            val logPath = context?.getExternalFilesDir("")
            val logNames = arrayOf("cellinfolte.csv", "ltecells.csv",
                    "esmrcells.csv", "cdmacells.csv", "gsmcells.csv")

            for (fileName in logNames) {
                val logFile = File(logPath, fileName)

                if (logFile.exists()) {
                    logFile.delete()
                }
            }

            Toast.makeText(context, text, duration).show()
            false
        }
    }
}
